core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -371,6 +374,11 @@ class SparkContext(config: SparkConf) extends Logging {
       throw new SparkException("An application name must be set in your configuration")
     }
 
+    _resources = conf.get(DRIVER_GPU_ADDRESSES).map(ids => {
```

Comment:
yes gpu's on the driver, I think the main use case is standalone mode or if someone doesn't have isolation. They can't just look on the host and take all the gpu's as they should only use the ones the cluster manager assigned to them. 
Yes it will be by convention but its better thenwhat we have now